unstructured/staging/base.py

```diff
@@ -216,10 +222,11 @@ def flatten_dict(
                 )
         else:
             flattened_dict[new_key] = value
+
     return flattened_dict
 
 
-def _get_table_fieldnames(rows):
+def _get_table_fieldnames(rows: list[dict[str, Any]]):
```

Comment:
Are you using `list` instead of `Sequence` because this is a private method?

Reply:
Hmm, no, this could definitely be `Sequence`. The main reason is the type checker wasn't complaining and we know it will always be this exact type and never `list[some-subtype-of-dict[str, Any]]`. But it wouldn't hurt to change it to `Sequence`.

The reason i use `Sequence` for elements is because `list` is invariant. Invariant means that this function call will type-check:
```python
def f(elements: list[Element]):
    ...

f([Element("abc"), Element("def")])
```
but not this one:
```python
f([Title("abc"), Text("def")])
```

The idea being that a list of `Element` _subtypes_ are not acceptable as a list of `Element`.

`Sequence` on the other hand is _covariant_, which means a sequence of `Element` subtypes is acceptable as `Sequence[Element]`.

The other kind of variance is contra-variance. More on that business in this article: https://www.playfulpython.com/type-hinting-covariance-contra-variance/

And I like this guy's videos on the subject: https://www.youtube.com/watch?v=7Chd5gPHlDg

A good principle is: "be as permissive as possible for input types and as specific as possible for return types". Following that principle, all we need of `rows` is `Iterable[dict[str, Any]]`. So if we were going to change it then probably that would be the best choice.

My intention in these initial mechanical refactorings has just been to get to clean-strict as quickly as possible, so I haven't thought a lot about that, but going forward, as long as we're putting in the effort, it probably makes sense to do as good a job as we can, I'm sure it doesn't take any longer really.